### PR TITLE
Improve Installation and Configuration for WP

### PR DESCRIPTION
**Pull request by joshcanhelp**
This group of commits helps to clarify and consolidate documentation for the Login by Auth0 WordPress plugin. Summary:

- Added a "Manual setup" section to the Installation page
- Added more information about Client setup (including adding new) Database Connections, and Social Connections; removed a duplicate section on WP hooks; reordered the settings fields
- Cleaned up the index page
- Added a few items from the wp.org readme that will be removed in the next version

https://auth0-docs-content-pr-5872.herokuapp.com/docs/cms/wordpress
https://auth0-docs-content-pr-5872.herokuapp.com/docs/cms/wordpress/configuration
https://auth0-docs-content-pr-5872.herokuapp.com/docs/cms/wordpress/troubleshoot

### CODE REVIEW

File: articles/cms/wordpress/configuration.md [hunk 1]
```diff
@@ -101,9 +92,30 @@ In order for your WordPress site to perform certain actions on behalf of your Au
 
 ![Client Advanced Settings](/media/articles/cms/wordpress/grant-client-access-to-api.png)
 
+### Database Connection setup
+
+Database Connections enable the typical username and password login seen on most sites. This type of Connection is 
+not required and can be skipped if you're using passwordless or social logins only. 
+
+1. Navigate to the [Connections > Database](${manage_url}/#/connections/database) page and look for a Connection that has a similar name to the Client setup above, if you used the wizard during setup. Otherwise, you can create a new
```
Comment (jeffreylees) [inline articles/cms/wordpress/configuration.md]: Maybe start with the condition "If you used the wizard during setup, navigate..."

Also maybe "use an existing Connection" rather than "use and existing Connection created previously" - an/and, and the extra seems extraneous.

File: articles/cms/wordpress/configuration.md [hunk 2]
```diff
@@ -101,9 +92,30 @@ In order for your WordPress site to perform certain actions on behalf of your Au
...
+not required and can be skipped if you're using passwordless or social logins only. 
+
+1. Navigate to the [Connections > Database](${manage_url}/#/connections/database) page and look for a Connection that has a similar name to the Client setup above, if you used the wizard during setup. Otherwise, you can create a new  
+Connection, use and existing Connection created previously, or use the default **Username-Password-Authentication**. Click an existing Connection name to view settings or click **Create DB Connection** and follow the steps.
+
+    ![Client Advanced Settings](/media/articles/cms/wordpress/database-connection-listing.png)
+
+1. Click the **Settings** tab, set **Password Strength** to the same as your wp-admin setting (default is Fair), and click **Save** at the bottom. If you want your password policy to be stronger or weaker, make sure to set it both here and at **wp-admin > Auth0 > Settings**.
+
+1. Now click the **Clients** tab and activate the Client created above
```
Comment (jeffreylees) [inline articles/cms/wordpress/configuration.md]: If you're writing complete sentences on your list items I'd punctuate the end.

File: articles/cms/wordpress/configuration.md [hunk 3]
```diff
@@ -101,9 +92,30 @@ In order for your WordPress site to perform certain actions on behalf of your Au
...
+
+    ![Client Advanced Settings](/media/articles/cms/wordpress/db-connection-clients.png)
+
+### Social Connection setup
+
+See our [dedicated page on Social Connections](/identityproviders#social) for detailed information on how to activate and configure these login 
+methods.
```
Comment (jeffreylees) [inline articles/cms/wordpress/configuration.md]: line break

File: articles/cms/wordpress/configuration.md [hunk 4]
```diff
@@ -161,71 +171,66 @@ In order for your WordPress site to perform certain actions on behalf of your Au
...
+* **Enable Gravatar Integration:** When user enters their email, their associated Gravatar picture is displayed in the Lock header.
 
-* **Lock primary color:** Information on this setting is [here](/libraries/lock/v11/configurationn#primarycolor-string-).
+* **Login Form CSS:** Valid CSS that will be applied to the login page. For more information on customizing Lock, see [Can I customize the Login Widget?](https://github.com/auth0/wp-auth0#can-i-customize-the-login-widget).
 
-* **Lock Language:** Information on this setting is [here](/libraries/lock/v11/configuration#language-string-).
+* **Login Form JS:** Valid JS that will be applied to the login page. For more information on customizing Lock, see [Can I customize the Login Widget?](https://github.com/auth0/wp-auth0#can-i-customize-the-login-widget).
 
-* **Lock Language Dictionary:** Information on this setting is [here](/libraries/lock/v11/configuration#languagedictionary-object-).
+* **Login Name Style:** Selecting **Email** will require users to enter their email address to login. Set this to *username* if you do not want to force a username to be a valid email address.
```
Comment (jeffreylees) [inline articles/cms/wordpress/configuration.md]: Bold **Username** to match **Email**  - match styles since you're referring to values for the same settings in the same sentence.

File: articles/cms/wordpress/configuration.md [hunk 5]
```diff
@@ -161,71 +171,66 @@ In order for your WordPress site to perform certain actions on behalf of your Au
...
 
-* **Remember last login:** Requests SSO data and enables the *Last time you signed in with[...]* option. For more information,  see [rememberLastLogin {Boolean}](/libraries/lock/customization#rememberlastlogin-boolean-).
+* **Primary Color:** Information on this setting is [here](/libraries/lock/v11/configurationn#primarycolor-string-).
```
Comment (jeffreylees) [inline articles/cms/wordpress/configuration.md]: extra n `configurationn`

File: articles/cms/wordpress/installation.md [hunk 6]
```diff
@@ -17,7 +15,20 @@ As soon as the plugin is activated, you are redirected to the start of the Setup
...
 If you don't already have an Auth0 account, click the **[Sign Up For Free](https://auth0.com/signup).** button and create one before proceeding. 
 
-## Social Login
+## Manual setup
+
+The plugin can be configured using the built-in setup wizard (covered below) or manually by creating a Client and 
+assigning connections. The completely manual setup process can be used if you're having trouble with the 
+wizard, have been through the setup process before, or want to share a database connection between Clients.
+
+1. In your Auth0 Dashboard, click **Clients** then **Create Client**
```
Comment (jeffreylees) [inline articles/cms/wordpress/installation.md]: You can link here `[Dashboard](${manage_url})`

File: articles/cms/wordpress/troubleshoot.md [hunk 7]
```diff
@@ -4,10 +4,22 @@ description: This page explains common troubleshooting issues with WordPress.
 
 # WordPress Troubleshooting
 
+## I'm seeing a message "This account does not have an email associated..." that stops me from logging in.
+
+If you get this error, make sure you are requesting an email from each provider in the Auth0 Dashboard under Connections -> Social (expand each provider). Take into account that not all providers return email addresses for users (e.g. Twitter). If this happens, you can always add an Email address to any logged in user through the Auth0 Dashboard (or API). See Users -> Edit.
+
+## I'm getting a "Failed cross origin authentication" or "No verifier returned from client" error in my error logs or when logging in.
+
+Check your "Allowed Callback URLs" and "Allowed Origins (CORS)" fields in the Client settings for your WordPress site to make sure those are correct. If you're using a Chromium-based browser, review our [docs page on cross-origin authentication](https://auth0.com/docs/cross-origin-authentication#limitations-of-cross-origin-authentication) to make sure you don't have third-party cookies turned off.
```
Comment (jeffreylees) [inline articles/cms/wordpress/troubleshoot.md]: Relative URL `[docs page on cross-origin authentication](/cross-origin-authentication#limitations-of-cross-origin-authentication")`

File: articles/cms/wordpress/troubleshoot.md [hunk 8]
```diff
@@ -4,10 +4,22 @@ description: This page explains common troubleshooting issues with WordPress.
...
+
 ## The Auth0 settings page in WordPress displays the warning: "The current user is not authorized to manage the Auth0 account...".
 
 If you updated your plugin to version 2 or configured the plugin without following the Quick Start Guide, you may need to provide an API token that the plugin will use to update your account settings. You can [generate a new token](/api/v2) and enter it into the **App Token** field of the the **Basic** settings page of the plugin. (The required scopes for the token are listed there.) You can also ignore this warning. Some operations will not be available from the plugin (like enabling rules or SSO). You will need to make these configuration changes manually in the [Auth0 dashboard](${manage_url}/#/clients).
 
+## I have two accounts for the same user in WordPress.
+
```
Comment (jeffreylees) [inline articles/cms/wordpress/troubleshoot.md:20]: WordPress allows you to merge users.

File: articles/cms/wordpress/troubleshoot.md [hunk 9]
```diff
@@ -70,6 +82,24 @@ The current version of the plugin does not provide a way to automatically migrat
 
 - Use the [WordPress XML RPC](https://codex.wordpress.org/XML-RPC_Support) endpoint to setup the migration flow using a custom database connection as described in [Import users to Auth0](/connections/database/migrating) with [this script](https://gist.github.com/glena/b31716e3c8fe48927be2).
 
+## The form_title setting is ignored when I set up the dict setting
+
+Internally, the plugin uses the dict setting to change the Auth0 widget title. When you set up the dict field it overrides the form_title one.
+
+To change the form_title in this case, you need to add the following attribute to the dict json:
+
+      {
```
Comment (jeffreylees) [inline articles/cms/wordpress/troubleshoot.md]: Code block this

**Review by jeffreylees — CHANGES_REQUESTED** [review]
Just copyediting changes, the content looks good to me

**Comment by joshcanhelp** [comment]
Thanks @jeffreylees, just committed fixes now. Let me know if there's anything else.

**Review by jeffreylees — APPROVED** [review]

**Comment by joshcanhelp** [comment]
@mpaktiti - First approval complete, ready for you!

File: articles/cms/wordpress/configuration.md [hunk 10]
```diff
@@ -25,7 +25,11 @@ You'll need to be logged into your Auth0 account before starting the steps below
 
 First, we'll check for the Client created for your WordPress site. 
 
-1. Navigate to the [Clients](${manage_url}/#/clients) page and look for a client that is similar to your site name; if you don't find one, it means that a Client was not created by the Wizard. Restart the Setup Wizard and try again.  
+1. Navigate to the [Clients](${manage_url}/#/clients) page and look for a Client that is similar to your site name; if you don't find one, it means that a Client was not created by the Wizard. Restart the Setup Wizard or create a new Client:
+    
+    1. Click **Create Client**
```
Comment (mpaktiti) [inline articles/cms/wordpress/configuration.md]: Maybe change this to a bulleted list? Looks kind of funny having step 1 under step 1

image

File: articles/cms/wordpress/configuration.md [hunk 11]
```diff
@@ -35,15 +39,15 @@ First, we'll check for the Client created for your WordPress site.
 
 1. **Client Type** must be set to **Regular Web Application**
```
Comment (mpaktiti) [inline articles/cms/wordpress/configuration.md:36]: You already say on the first step to select regular web app as type, so we can remove this.

**Review by mpaktiti — CHANGES_REQUESTED** [review]
Looks good Josh, I added a couple of comments for minor things

File: articles/cms/wordpress/configuration.md (hunk 10 above)
Comment (joshcanhelp) [reply articles/cms/wordpress/configuration.md]: Bullet points are for non-sequential items, where order doesn't matter. I could combine into the paragraph separated by commas?

File: articles/cms/wordpress/configuration.md (hunk 11 above)
Comment (joshcanhelp) [reply articles/cms/wordpress/configuration.md:36]: Where I say it on step 1 is if someone is creating a new client. If they're troubleshooting an existing one then this needs to match.

File: articles/cms/wordpress/configuration.md (hunk 10 above)
Comment (mpaktiti) [reply articles/cms/wordpress/configuration.md]: yeap, sounds good

File: articles/cms/wordpress/configuration.md (hunk 11 above)
Comment (mpaktiti) [reply articles/cms/wordpress/configuration.md:36]: In this case I'd expect to see that in a Troubleshooting section, under the relevant error

Comment (joshcanhelp) [reply articles/cms/wordpress/configuration.md:36]: To provide a little more context here ... the Clients created when WordPress is installed are created as they need to be when the installer is run. The problem we run across is that Clients created a year ago might need to be updated in a few ways (this attribute being one of them). We try to do this automatically but sometimes that process fails and the updates need to be done manually. If that's the case, we direct folks to this Configuration page and say "make sure your client matches this." The errors that they experience run the gamut, in fact it would be incredibly difficult to know what, exactly, they would see in each case. If someone is reading this, they skip the part about creating a new Client and, if this is missing, they skip that configuration as well. 

TL; DR: the repetition is not harmful, IMHO, and is necessary in a very common support use case.

File: articles/cms/wordpress/configuration.md (hunk 10 above)
Comment (joshcanhelp) [reply articles/cms/wordpress/configuration.md]: @mpaktiti - OK, changed!

File: articles/cms/wordpress/configuration.md (hunk 11 above)
Comment (joshcanhelp) [reply articles/cms/wordpress/configuration.md:36]: @mpaktiti - Any thoughts on this?

Comment (mpaktiti) [reply articles/cms/wordpress/configuration.md:36]: I was missing the context, it makes sense now. Sorry for the delay, I'm merging now.
